src/components/molecules/Weather.tsx

```diff
@@ -47,40 +41,26 @@ export default function Weather(props: Props) {
     setDay(`${dayOfWeekStr[dayOfWeek]}, ${month} ${day}, ${year}`);
   }, []);
 
-  // `${SERVER_URL}/en/1`
-  useEffect(() => {
-    axios
-      .get(
-        `${WEATHER_URL}?lat=${props.lat}&lon=${props.long}&units=metric&appid=${process.env.NEXT_PUBLIC_WEATHER_API}`
-      )
-      .then((res) => {
-        setData(res.data);
-      });
-  }, [props.lat, props.long]);
 
-  console.log("Latitude is:", props.lat);
-  console.log("Longitude is:", props.long);
+  const WEATHER_URL = data.weather_icon
+    ? `http://openweathermap.org/img/wn/${data.weather_icon}.png`
+    : "";
 
   return (
     <div className={styles.weather}>
-      {typeof data.main != "undefined" ? (
+      {typeof data != "undefined" ? (
```

Comment:
Is the `data` will be `undefined`?
Because this `Weather` component is used by `Top.tsx` but It will render this component when the `data` is something.
```tsx
{data ? <Weather data={data} /> : <></>}
```